app/(dashboard)/dashboard/pull-request.tsx

```diff
@@ -70,6 +70,9 @@ export function PullRequestItem({ pullRequest: initialPullRequest }: PullRequest
   const { toast } = useToast();
   const [commitMessage, setCommitMessage] = useState("Update test files");
 
+  const [streamedTestFiles, setStreamedTestFiles] = useState<TestFile[]>([]);
```

Comment:
This seems a bit boilerplate-y and I think we'd get this for free with streamObjects?

A video comparison would also help

Reply:
I totally agree with you and this is not my preferred way of doing it. I think streaming should happen in the generate-test route and handle it in pull-request. However, I temporarily made these changes because the current implementation is glitchy and in some cases, it fails to handle exceptions properly and causes the app to crash (unfortunately was not able to show this in the video but I faced this issue a couple times). 

This is boilerplate-y but it's reliable, however, if you don't feel comfortable with this, it would be better to revert streaming test files for now in my opinion. 

Quick side by side comparison (left one being this branch) 

https://github.com/user-attachments/assets/70260b7a-f72b-45b0-86d4-da7552e58a85